pkgs/os-specific/darwin/cctools/port.nix

```diff
@@ -37,9 +37,12 @@ stdenv.mkDerivation {
 
   nativeBuildInputs = [ autoconf automake libtool autoreconfHook installShellFiles ];
   buildInputs = [ libuuid ]
+    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ memstream ]
```

Comment:
Is it possible to put `memstreamHook` here to avoid appending `env.NIX_LDFLAGS`?

Reply:
I can do that. Since the hook also supplies the header, I’ve changed the patch to unconditionally use `open_memstream`. I’ll be pushing an updated commit once I’ve confirmed that it builds and works.